cli/kubernetes/options_test.go

```diff
@@ -17,6 +19,67 @@ var _ = Describe("InstallationOption", func() {
 			Expect(option.ToOptMapKey()).To(Equal("TheName-SomeDeployment"))
 		})
 	})
+
+	Describe("DynDefault", func() {
+		option := InstallationOption{
+			Name:         "TheName",
+			DeploymentID: "SomeDeployment",
+			DynDefaultFunc: func(o *InstallationOption) error {
+				o.Value = "Hello"
+				return nil
+			},
+		}
+		It("it calls the DynDefaultFunc", func() {
+			Expect(option.DynDefault()).To(BeNil())
+			Expect(option.Value).To(Equal("Hello"))
+		})
+	})
+
+	Describe("SetDefault", func() {
+		// The tests here are the origin for the tests of
+		// `DefaultOptionsReader.Read` in files
+		// `default_options_reader(_test).go`.  Any new tests
+		// here should be replicated there.
+
+		optionDynamic := InstallationOption{
+			Name:         "TheName",
+			DeploymentID: "SomeDeployment",
+			Default:      "World",
+			DynDefaultFunc: func(o *InstallationOption) error {
+				o.Value = "Hello"
+				return nil
+			},
+		}
+		optionStatic := InstallationOption{
+			Name:         "TheName",
+			DeploymentID: "SomeDeployment",
+			Default:      "World",
+		}
+		optionError := InstallationOption{
+			Name:         "TheName",
+			DeploymentID: "SomeDeployment",
+			DynDefaultFunc: func(o *InstallationOption) error {
+				o.Value = "Hello"
+				return errors.New("an error")
+			},
+		}
+
+		It("it prefers the DynDefaultFunc over a static Default", func() {
```

Comment:
[minor] I think the "convention" is `It("prefers..."` (without an `it` in the text). In other places as well.

Reply:
Yes. Tried to follow it, missed these. Will fix.

Reply:
Fixed with commit `6a70a803f0b186c609364a3da9550d393a338e95`